### ISSUE TITLE
Feature Request: Implement a Model ID column in the results from `tune_grid()`

### ISSUE THREAD

**Issue by mdancho84** [tidymodels/tune#209]
Love the `tune_grid()` function. The output wrangling functions `show_best()` and `select_best()` could be enhanced by making it easier to select specific models. 

## Enhancement Request

An enhancement is to include a Model ID and selection interface:

- A Model ID (`.model_id`) column in the results, which enhances the selection process.
- Implement `select_model_id(model_id = "xyz")` to allow the user to quickly select parameters for a Model ID.

## Why Model ID?

It's a good idea to have a unique identifier for parameter combinations. This can make it easier for users to select the model they want to use models other than those with the best mean error metric. Sometimes:

- The best mean error has higher standard error and is therefore has more variability
- Has failed several tuning combinations and is therefore less robust to new data.

## Other Implementations

The `H2O` R package has `h2o::getModel()`, which is very similar to what I'm proposing. It makes it super easy to get specific models with better or worse performance. It's used mainly when doing grid search or selecting models from the AutoML results (`h2o::h2o.automl()`).

**Comment by topepo** [comment]
Sort of like #87? That issue doesn't have much detail but this is the general idea.

**Comment by mdancho84** [comment]
Yep, exactly.

**Comment by juliasilge** [comment]
Closed in #227

The model ID column is called `.config`:

``` r
library(tidymodels)
#> ── Attaching packages ─────────────────────────────────────────────────────────────────── tidymodels 0.1.0 ──
#> ✓ broom     0.5.6          ✓ recipes   0.1.12    
#> ✓ dials     0.0.7          ✓ rsample   0.0.7     
#> ✓ dplyr     1.0.0          ✓ tibble    3.0.1     
#> ✓ ggplot2   3.3.1          ✓ tune      0.1.0.9000
#> ✓ infer     0.5.2          ✓ workflows 0.1.1.9000
#> ✓ parsnip   0.1.1.9000     ✓ yardstick 0.0.6.9000
#> ✓ purrr     0.3.4
#> ── Conflicts ────────────────────────────────────────────────────────────────────── tidymodels_conflicts() ──
#> x purrr::discard() masks scales::discard()
#> x dplyr::filter()  masks stats::filter()
#> x dplyr::lag()     masks stats::lag()
#> x recipes::step()  masks stats::step()

set.seed(6735)
folds <- vfold_cv(mtcars, v = 5)

spline_rec <-
  recipe(mpg ~ ., data = mtcars) %>%
  step_ns(disp, deg_free = tune("disp")) %>%
  step_ns(wt, deg_free = tune("wt"))

lasso_spec <-
  linear_reg(penalty = tune(), mixture = 1) %>%
  set_engine("glmnet")

lasso_wf <-
  workflow() %>%
  add_model(lasso_spec) %>%
  add_recipe(spline_rec)

lasso_res <- tune_grid(
  lasso_wf, 
  resamples = folds, 
  grid = 7, 
  control = control_grid(save_pred = TRUE)
)

collect_metrics(lasso_res)
#> # A tibble: 14 x 9
#>     penalty  disp    wt .metric .estimator   mean     n std_err .config       
#>       <dbl> <int> <int> <chr>   <chr>       <dbl> <int>   <dbl> <chr>         
#>  1 1.15e-10     4    11 rmse    standard   11.9       5  5.44   Recipe1_Model1
#>  2 1.15e-10     4    11 rsq     standard    0.462     5  0.173  Recipe1_Model1
#>  3 7.05e- 5    15     7 rmse    standard    6.56      5  3.75   Recipe2_Model1
#>  4 7.05e- 5    15     7 rsq     standard    0.684     5  0.0880 Recipe2_Model1
#>  5 3.82e- 1     1    14 rmse    standard    2.92      5  0.316  Recipe3_Model1
#>  6 3.82e- 1     1    14 rsq     standard    0.673     5  0.0757 Recipe3_Model1
#>  7 4.88e- 6     9     6 rmse    standard    4.80      5  0.669  Recipe4_Model1
#>  8 4.88e- 6     9     6 rsq     standard    0.387     5  0.169  Recipe4_Model1
#>  9 1.67e- 3    11     3 rmse    standard    3.80      5  0.801  Recipe5_Model1
#> 10 1.67e- 3    11     3 rsq     standard    0.598     5  0.151  Recipe5_Model1
#> 11 7.76e- 8     6     2 rmse    standard    2.89      5  0.330  Recipe6_Model1
#> 12 7.76e- 8     6     2 rsq     standard    0.671     5  0.124  Recipe6_Model1
#> 13 1.09e- 8     8    12 rmse    standard    3.13      5  0.533  Recipe7_Model1
#> 14 1.09e- 8     8    12 rsq     standard    0.611     5  0.136  Recipe7_Model1
collect_predictions(lasso_res)
#> # A tibble: 224 x 8
#>    id    .pred  .row  disp    wt  penalty   mpg .config       
#>    <chr> <dbl> <int> <int> <int>    <dbl> <dbl> <chr>         
#>  1 Fold1  17.5     7     4    11 1.15e-10  14.3 Recipe1_Model1
#>  2 Fold1  19.0    11     4    11 1.15e-10  17.8 Recipe1_Model1
#>  3 Fold1  14.9    14     4    11 1.15e-10  15.2 Recipe1_Model1
#>  4 Fold1  16.5    15     4    11 1.15e-10  10.4 Recipe1_Model1
#>  5 Fold1  35.0    19     4    11 1.15e-10  30.4 Recipe1_Model1
#>  6 Fold1  24.8    29     4    11 1.15e-10  15.8 Recipe1_Model1
#>  7 Fold1  11.2    31     4    11 1.15e-10  15   Recipe1_Model1
#>  8 Fold1  15.6     7    15     7 7.05e- 5  14.3 Recipe2_Model1
#>  9 Fold1  19.8    11    15     7 7.05e- 5  17.8 Recipe2_Model1
#> 10 Fold1  17.6    14    15     7 7.05e- 5  15.2 Recipe2_Model1
#> # … with 214 more rows
```

<sup>Created on 2020-06-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>
